views/media/edit.tt

```diff
@@ -9,43 +9,107 @@ This program is free software; you may redistribute it and/or modify it under
+<div id="media-list">
+    [% first_index = (page - 1) * 20 %]
+    [% last_index = page * 20 - 1 < media.size ? page * 20 - 1 : media.max %]
+    [%- FOREACH obj IN media.slice(first_index, last_index) -%]
+        
+        <div class='media-item [% loop.count % 2 ? 'even' : 'odd' %] row'>
         <div class='inner'>
-            <div class='media'><img src="[% obj.url %]" /></div>
-            <div class='details'><span class='name '>[%obj.displayid%]</span>
-                <br /><span class='filesize'>[% obj.size / 1000 %] kb</span></div>
-            [%- form.select( name="security-$obj.displayid"
-                items = security
-                selected = obj.security ) -%]
-            <div><label for="delete_[% obj.displayid %]">Delete?</label>
-                <input type="checkbox" name="delete" id="delete_[% obj.displayid %]" value="[% obj.displayid %]" />
+            <div class='media large-3 column'><img src="[% obj.url %]" /></div>
+            <div class='edit large-9 column'>
+            <div class="row">
+                    <div class="large-8 columns">
+                        [%- INCLUDE 'media/field-row.tt' 
+                            label = dw.ml(".edit.title")
+                            type = "text" 
+                            id = "edit-title-$obj.displayid" 
+                            value = obj.prop('title') 
+                            name = "title-$obj.displayid" 
+                            maxlength = '125'
+                            labelSize = 3 
+                        -%]
+                    </div>
+                    <div class="large-4 columns">
+                        [%- INCLUDE 'media/field-row.tt' 
+                            label = dw.ml(".edit.security")
+                            type = "select"
+                            name="security-$obj.displayid"
+                            items = security
+                            selected = obj.security 
+                            labelSize = 4
+                        -%]
+                    </div>
+                </div>
+            [% INCLUDE 'media/field-row.tt'  
+                label = dw.ml(".edit.alt")
+                type = "text" 
+                id = "edit-alttext-$obj.displayid" 
+                value = obj.prop('alttext') 
+                name = "alttext-$obj.displayid" 
+                maxlength = '125'
+            %]
+            [% INCLUDE 'media/field-row.tt' 
+                label = dw.ml(".edit.desc") 
+                type = "textarea"
+                id = "edit-description-$obj.displayid" 
+                value = obj.prop('description') 
+                name = "description-$obj.displayid"
+            %]
+            [% INCLUDE 'media/field-row.tt' 
+                type = "checkbox"
+                label = dw.ml(".edit.delete") 
+                id = "delete_$obj.displayid" 
+                name = "delete-$obj.displayid"
+            %]
```

Comment:
Ah ha, the old checkbox also had `value="[% obj.displayid %]` and I think that's what's causing both the undefined argument error and the failure to actually delete.